### PR TITLE
[JENKINS-64291] Fix drag & drop issues replacing the drag & drop library

### PR DESCRIPTION
**Pull request by fqueiruga**
See [JENKINS-64291](https://issues.jenkins-ci.org/browse/JENKINS-64291).

I haven't found a way to fix JENKINS-64291 other than straight up replacing the drag & drop library. This PR replaces the YUI drag & drop with https://github.com/SortableJS

Where to test:

- JDK and Git configuration on Global Tools configuration

#### TODOs

- [x] Ensure changes work with all plugins
- [x] Move the JS code to webpack and import the library through NPM
- [x] Refactor the new CSS customization code
- [x] Remove the legacy dragdrop.js file and CSS code
- [x] Stop importing the dragdrop.js and the YUI dragdrop module
- [x] Enable AutoScroll
- [x] Test on IE11

### Proposed changelog entries

* Replace the drag & drop library used by Jenkins with SortableJS. This fixes the drag & drop functionality for the form changes.

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [x] (If applicable) Jira issue is well described
- [ ] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@timja
@oleg-nenashev 
@Wadeck 
@halkeye 

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

**Review by timja — COMMENTED** [review]
I've tested this functionally and I can't find any issues, so I vote let's replace the old yui library.

Cleanup needed here first before doing that though

Thanks!

File: core/src/main/resources/lib/layout/layout.jelly
Comment (daniel-beck) [inline core/src/main/resources/lib/layout/layout.jelly:187]: Won't work in offline situations; we must not forget to change this.

File: war/src/main/js/widgets/config/tabbar.less
Comment (halkeye) [inline war/src/main/js/widgets/config/tabbar.less:28]:
```suggestion
     text-shadow:none;
```

**Comment by halkeye** [comment]
I'm really excited for any removal of YUI. Lemme know how I can help.

**Comment by fqueiruga** [comment]
Thank you Gavin. My manual testing so far has been successful. I did some code search across the plugin ecosystem and it seems some files may be sensible to these changes. I'd appreciate any help or indications for testing them out.

I'm listing only big profile plugins here

- scm-api
    - https://github.com/jenkinsci/scm-api-plugin/blob/master/src/main/resources/jenkins/scm/api/form/traits.jelly (`repeat-chunk`)
- branch-api
    - https://github.com/jenkinsci/branch-api-plugin/blob/master/src/main/resources/lib/branch-api/branchSourceSelector.jelly (`repeated-chunk`)
    - https://github.com/jenkinsci/branch-api-plugin/blob/master/src/main/resources/lib/branch-api/branchSourceBody.jelly (`dd-handle`) 
- envinject
    - https://github.com/jenkinsci/envinject-plugin/blob/master/src/main/resources/util/hetero-list-readonly.jelly  (custom `hetero-list`!!!)
- pipeline-model-definition (these may not be problematic, as per @timja 's comment on [JENKINS-62808](https://issues.jenkins.io/browse/JENKINS-62808))
    - https://github.com/jenkinsci/pipeline-model-definition-plugin/blob/master/pipeline-model-definition/src/main/resources/org/jenkinsci/plugins/pipeline/modeldefinition/generator/DirectiveGenerator/symbol-hetero-list.jelly
    - https://github.com/jenkinsci/pipeline-model-definition-plugin/blob/master/pipeline-model-definition/src/main/resources/org/jenkinsci/plugins/pipeline/modeldefinition/generator/ToolsDirective/config.jelly  
    
Pending clarifying these uses of custom elements, I think this PR is ready to accept reviews.

**Comment by timja** [comment]
There's 6 tests failing all seem to be for the same cause: https://github.com/jenkinsci/jenkins/pull/5177/checks?check_run_id=1727548397

**Comment by fqueiruga** [comment]
It may have to do with the removal of the YUI drag & drop module, that it affects the YUI container resizing

**Comment by fqueiruga** [comment]
I restored the YUI dragdrop module to see whether it still loads.

**Comment by MRamonLeon** [comment]
Can you keep drag&drop disabled while resizing enabled? To move forward removing YUI. Not sure whether YUI is structured such a way to allow that though.

**Comment by fqueiruga** [comment]
TBH I don't know what the YUI resizing module even does, and I don't know how risky it would be to break it. 

Given that the main purpose of this PR is a fix for an urgent issue, I'd suggest moving ahead  without removing the load of that YUI dragdrop module. We can do that as part of a follow-up ticket.

**Review by timja — APPROVED** [review]

**Review by escoem — APPROVED** [review]
Tested manually without issues.

**Review by res0nance — APPROVED** [review]

**Comment by timja** [comment]
This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Comment by daniel-beck** [comment]
Caused https://issues.jenkins.io/browse/JENKINS-67496

**Pull request merged**
